content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md

```diff
@@ -7,81 +7,86 @@ description: Creating Secret objects using resource configuration file.
-
-```
-MWYyZDFlMmU2N2Rm
-```
-
-Write a Secret config file that looks like this:
-
-```yaml
-apiVersion: v1
-kind: Secret
-metadata:
-  name: mysecret
-type: Opaque
-data:
-  username: YWRtaW4=
-  password: MWYyZDFlMmU2N2Rm
-```
-
-Note that the name of a Secret object must be a valid
-[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
-
-{{< note >}}
-The serialized JSON and YAML values of Secret data are encoded as base64
-strings. Newlines are not valid within these strings and must be omitted. When
-using the `base64` utility on Darwin/macOS, users should avoid using the `-b`
-option to split long lines. Conversely, Linux users *should* add the option
-`-w 0` to `base64` commands or the pipeline `base64 | tr -d '\n'` if the `-w`
-option is not available.
-{{< /note >}}
-
-For certain scenarios, you may wish to use the `stringData` field instead. This
-field allows you to put a non-base64 encoded string directly into the Secret,
-and the string will be encoded for you when the Secret is created or updated.
-
-A practical example of this might be where you are deploying an application
-that uses a Secret to store a configuration file, and you want to populate
-parts of that configuration file during your deployment process.
-
-For example, if your application uses the following configuration file:
+## Create a Secret {#create-the-config-file}
+
+You can define the `Secret` object in a JSON or YAML configuration file before
+you create the object. The [Secret](/docs/reference/generated/kubernetes-api/{{<
+param "version" >}}/#secret-v1-core) resource contains two maps: `data` and
+`stringData`.
+
+The `data` field stores arbitrary base64-encoded data. The `stringData` field
+lets you provide the same data as unencoded strings. The keys in the `data` and
+`stringData` must consist of alphanumeric characters, `-`, `_`, or `.`.
+
+The following example creates a Secret that stores the username `admin` and the
+password `VLc5@#tym$HzBn@8`.
+
+1.  Encode the data:
+
+    ```shell
+    echo -n 'admin' | base64
+    echo -n 'VLc5@#tym$HzBn@8' | base64
+    ```
+    The `-n` flag ensures that there's no newline character at the end of your
+    data. If you're using the `base64` utility on Darwin/macOS, avoid using the
+    `-b` option to split long lines. Conversely, if you're on Linux, add the
+    `-w 0` option to `base64` commands or the pipeline `base64 | tr -d '\n'` if
+    the `-w` option is not available.
+
+    The output is similar to:
+
+    ```
+    YWRtaW4=
+    VkxjNUAjdHltJEh6Qm5AOA==
+    ```
+
+1.  Create the configuration file:
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: mysecret
+    type: Opaque
+    data:
+      username: YWRtaW4=
+      password: VkxjNUAjdHltJEh6Qm5AOA==
+    ```
+
+    The name of the `Secret` object must be a valid
+    [DNS subdomain
+    name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
```

Comment:
Please keep hyperlinks on the same line. We are still unable to parse multiline links when identifying dangling ones.